app/assets/javascripts/discourse/app/mixins/composer-video-thumbnail-uppy.js

```diff
@@ -21,76 +21,86 @@ export default Mixin.create(ExtendableUploader, UppyS3Multipart, {
     let video = document.createElement("video");
     video.src = URL.createObjectURL(videoFile.data);
 
+    // These attributes are needed for thumbnail generation on mobile.
+    // This video tag is not visible, so this is all happening in the background.
+    video.autoplay = true;
+    video.muted = true;
+    video.playsinline = true;
+
     let videoSha1 = uploadUrl
       .substring(uploadUrl.lastIndexOf("/") + 1)
       .split(".")[0];
 
-    // Wait for the video element to load, otherwise the canvas will be empty
-    video.oncanplay = () => {
+    // Wait for the video element to load, otherwise the canvas will be empty.
+    // iOS Safari perfers onloadedmetadata over oncanplay.
```

Comment:
```suggestion
    // iOS Safari prefers onloadedmetadata over oncanplay.
```

Reply:
It is normally said "perfers" actually... kinda makes you wonder doesn't it.